src/main/java/flight/reservation/order/FlightOrder.java

```diff
@@ -1,20 +1,33 @@
 package flight.reservation.order;
 
 import flight.reservation.Customer;
+import flight.reservation.Passenger;
 import flight.reservation.flight.ScheduledFlight;
 import flight.reservation.payment.CreditCard;
 import flight.reservation.payment.Paypal;
 
 import java.util.Arrays;
 import java.util.Date;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class FlightOrder extends Order {
     private final List<ScheduledFlight> flights;
     static List<String> noFlyList = Arrays.asList("Peter", "Johannes");
 
-    public FlightOrder(List<ScheduledFlight> flights) {
+    public FlightOrder(Customer customer, List<String> passengerNames, List<ScheduledFlight> flights, double price) {
```

Comment:
consider to simplify the constructor and move setCustomer, setPrice, etc. to another method?